roles/ceph-osd/tasks/activate_osds.yml

```diff
@@ -69,7 +69,7 @@
     is_before_infernalis
 
 - name: get osd id (for or after infernalis)
-  shell: "ls /var/lib/ceph/osd/ |grep -oh '[0-9]*'"
+  shell: "ls /var/lib/ceph/osd/ | grep '-' | cut -d '-' -f 2"
```

Comment:
what if there are non-dashed files though? the grep ensures just dashed files will be used:


    $ touch foo bar osd-1 osd-2
    $ ls . | cut -d '-' -f 2
    bar
    foo
    1
    2
    $ ls . | grep '-' | cut -d '-' -f 2
    1
    2